api/views/content.py

```diff
@@ -155,13 +195,20 @@ class PublicPostsViewSet(
     renderer_classes = (PostsJSONRenderer,)
 
     def list(self, request):
-        internal_posts = Post.objects.filter(visibility="PUBLIC")
-        serializer = PostSerializer(internal_posts, many=True)
-        posts = serializer.data
+        return Response(get_public_posts(self, request))
+
+
+def get_public_posts(self, request):
+    """
+    Returns a list of public posts based on APIUser type.
+    """
+    internal_posts = Post.objects.filter(visibility="PUBLIC")
+    serializer = PostSerializer(internal_posts, many=True)
+    posts = serializer.data
 
-        # dont return public posts of other nodes in node-to-node calls
-        if request.auth is None or request.user.type is not "Node":
-            foreign_posts = Aggregator.get_public_posts()
-            posts.extend(foreign_posts)
+    # dont return public posts of other nodes in node-to-node calls
+    if request.auth is None or request.user.type is not "Node":
+        foreign_posts = Aggregator.get_public_posts()
+        posts.extend(foreign_posts)
 
-        return Response(posts)
+    return posts
```

Comment:
Smushed the separated function back into the view. I'm just going to pull out the JSONRendered because it's confusing.